logback/shared/src/main/scala/monix/mdc/MonixMDCAdapter.scala

```diff
@@ -59,4 +61,17 @@ object MonixMDCAdapter {
     field.set(null, new MonixMDCAdapter)
     field.setAccessible(false)
   }
+
+  def productToMap[P <: Product](product: P): Map[String, String] = {
+    val fields = product.getClass.getDeclaredFields.map(_.getName)
+    val values = product.productIterator.toSeq.map {
+      case maybe: Option[_] => maybe
+      case value => Option(value)
+    }
+    fields.zip(values).collect{ case (k, Some(v)) => (k, v.toString)}.toMap
+  }
+
+  def updateContext[P <: Product](product: P): Unit = {
```

Comment:
I think it would be nice to add a small scaladoc explaining what it does

Reply:
Done!